server/status_test.go

```diff
@@ -273,7 +275,41 @@ func TestNodeStatusResponse(t *testing.T) {
 	if len(nodeStatuses) != 1 {
 		t.Errorf("too many node statuses returned - expected:1 actual:%d", len(nodeStatuses))
 	}
-	if !reflect.DeepEqual(ts.node.Descriptor, nodeStatuses[0].Desc) {
-		t.Errorf("node status descriptors are not equal\nexpected:%+v\nactual:%+v\n", ts.node.Descriptor, nodeStatuses[0].Desc)
+	if !reflect.DeepEqual(tsrv.node.Descriptor, nodeStatuses[0].Desc) {
+		t.Errorf("node status descriptors are not equal\nexpected:%+v\nactual:%+v\n", tsrv.node.Descriptor, nodeStatuses[0].Desc)
 	}
 }
+
+func TestMetricsRecording(t *testing.T) {
+	tsrv := &TestServer{}
+	tsrv.Ctx = NewTestContext()
+	tsrv.Ctx.MetricsFrequency = 5 * time.Millisecond
+	if err := tsrv.Start(); err != nil {
+		t.Fatal(err)
+	}
+	defer tsrv.Stop()
+
+	// Verify that metrics for the current timestamp are recorded. This should
+	// be true very quickly.
+	util.SucceedsWithin(t, time.Second, func() error {
+		now := tsrv.Clock().PhysicalNow()
+		key := ts.MakeDataKey("cr.store.livebytes.1", "", ts.Resolution10s, now)
+		call := client.Get(key)
```

Comment:
We do, but I also created the proto.InternalTimeSeriesDataFromValue() which also verifies that the Value has the correct data tag - the tag is necessary for the C++ merge logic to identify time series data.